packages/internal/src/diagnostics/index.ts

```diff
@@ -0,0 +1,137 @@
+import { Project, ts } from 'ts-morph';
+import { init, getGraphQLDiagnostics } from '@0no-co/graphqlsp/api';
+import path from 'path';
+import fs from 'fs';
+import { resolveTypeScriptRootDir } from '../resolve';
+import type { SchemaOrigin } from '../loaders';
+import { load } from '../loaders';
+
+type Severity = 'error' | 'warn' | 'info';
+const severities: Severity[] = ['error', 'warn', 'info'];
+export interface FormattedDisplayableDiagnostic {
+  severity: Severity;
+  message: string;
+  line: number;
+  col: number;
+  file: string | undefined;
+}
+
+type GraphQLSPConfig = {
+  name: string;
+  schema: SchemaOrigin;
+  tadaOutputLocation: string;
+};
+
+export async function check(
+  config: GraphQLSPConfig,
+  minSeverity: Severity = 'error'
+): Promise<FormattedDisplayableDiagnostic[]> {
+  const projectName = path.resolve(process.cwd(), 'tsconfig.json');
+
+  const project = new Project({
+    tsConfigFilePath: './tsconfig.json',
+  });
```

Comment:
Question (not a blocker): I'm a bit concerned about this being hardcoded to `./tsconfig.json`. And the same goes for other parts of the CLI using mainly `${CWD}/tsconfig.json`.

I tested `new Project()` and it seems to indeed not trigger the resolver, so todo here to figure out resolver, which could be grouped with the polishing task to figure out reusing `TsConfigResolver` from `@ts-morph/common`, and then to also get to replacing `resolveTypeScriptRootDir` with something externally maintained, as discussed 🤔 ❤️